lib/node_modules/@stdlib/stats/strided/distances/dcosine-similarity/lib/ndarray.js

```diff
@@ -0,0 +1,70 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var ddot = require( '@stdlib/blas/base/ddot' ).ndarray;
+var dnrm2 = require( '@stdlib/blas/base/dnrm2' ).ndarray;
+
+
+// MAIN //
+
+/**
+* Computes the cosine similarity of two double-precision floating-point vectors using alternative indexing semantics.
+*
+* @param {integer} N - number of indexed elements
```

Comment:
```suggestion
* @param {NonNegativeInteger} N - number of indexed elements
```

This is more to signal that users should not be providing negative integers, as they are likely the result of an error in upstream computation.